### ISSUE TITLE
Build fails on Visual Studio 14 2015

### ISSUE THREAD

**Issue by katzi77** [protocolbuffers/protobuf#7436]
**What version of protobuf and what language are you using?**
Version: master/v3.11.2
Language: C++

**What operating system: Windows 10

**What runtime / compiler are you using (e.g., python version or gcc version)**
-- Selecting Windows SDK version  to target Windows 10.0.14393.
-- The C compiler identification is MSVC 19.0.23026.0
-- The CXX compiler identification is MSVC 19.0.23026.0

Hello,
Compiling Protobuf fails on MSVC 14 2015.
The same error appears multiple times:
Error	C2228	left of '.GetArena' must have class/struct/union (compiling source file C:\protobuf-3.11.2\src\google\protobuf\compiler\cpp\cpp_service.cc)	libprotoc	C:\protobuf-3.11.2\src\google\protobuf\arena.h	441

Using CMAKE:
cmake -G"Visual Studio 14 2015 Win64" -Dprotobuf_BUILD_SHARED_LIBS=ON -Dprotobuf_BUILD_TESTS="OFF"  -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=../../../Bin/Windows/VC14/Release  ../..

**Comment by thornicthorny** [comment]
Do you have the solution for this problem? I got the same error.

**Comment by fowles** [comment]
3.11 is a very old release that we no longer support.  Does this reproduce on more modern releases?

**Comment by thornicthorny** [comment]
Yes. I was building it from vcpkg. Version of protobuf is 3.19.4. Visual studio version is 2015 update 3.

**Comment by MartinChan3** [comment]
I have the same problem and it is very wired...Only on Windows with MSVC2015

**Comment by thornicthorny** [comment]
@MartinChan3  Yes I already solve the problem but thanks for the info. Previously I got the build error with VS2015 (without update) and then I updated to update 3 but got the same error. After I posted issue, I cleaned up all the interim files and restarted everything from a clean vcpkg and it worked.

**Comment by fowles** [comment]
Sounds like this was a MSVC bug then
